src/parachain/issue.ts

```diff
@@ -207,16 +192,29 @@ export class DefaultIssueAPI extends DefaultTransactionAPI implements IssueAPI {
 
     async request(
         amount: MonetaryAmount<WrappedCurrency, BitcoinUnit>,
-        vaultId?: AccountId,
+        vaultAccountId?: AccountId,
+        collateralCurrencyIdLiteral?: CurrencyIdLiteral,
         atomic: boolean = true,
         retries: number = 0,
-        cachedVaults?: Map<AccountId, MonetaryAmount<WrappedCurrency, BitcoinUnit>>
+        cachedVaults?: Map<InterbtcPrimitivesVaultId, MonetaryAmount<WrappedCurrency, BitcoinUnit>>,
     ): Promise<Issue[]> {
         try {
-            if (vaultId) {
+            if (vaultAccountId) {
+                if (!collateralCurrencyIdLiteral) {
+                    return Promise.reject(
+                        new Error("A collateral currency must be specified along with the vault account ID")
```

Comment:
Not sure what is best-practice (maybe @tomjeatt has a recommendation) but I'm not a huge fan of in-line error messages.

Reply:
Good point - libs aren't something I've had much experience with, but in a client app I would use a dictionary for handling errors.

Reply:
We have quite a few of these (here too https://github.com/interlay/interbtc-api/blob/master/src/clients/client.ts#L51). I'll add to backlog